src/main/resources/db/migration/postgres/V2.4.1.0__data_column_order_states.sql

```diff
@@ -0,0 +1,338 @@
+		select e.protocol_id,
+			coalesce(col_type.code_value, col_type.string_value) as column_type, --the coalesce is used to account for having the same data in stringValues or in codeValues
+			coalesce(col_name.code_value, col_name.string_value) as column_name,
+			coalesce(col_name.code_value, col_name.string_value) as units,
+			col_conc.numeric_value as concentration,
+		 	coalesce(col_conc_units.code_value, col_conc_units.string_value) as conc_units,
+			coalesce(hide_col.code_value, hide_col.string_value) as hide_column,
+			coalesce(cond_col.code_value, cond_col.string_value) as condition_column,
+			--This query will GROUP BY all the columns above this point
+			-- For column order and experiment, we take the lowest values as the same endpoint can show up in many experiments
+			min(col_order.numeric_value) as column_order,
+			min( e.id) as experiment_id
+		from experiment e
+		join experiment_state es on es.experiment_id = e.id and es.ignored = false and es.ls_kind = 'data column order'
+		join experiment_value col_type on col_type.experiment_state_id = es.id and col_type.ls_kind = 'column type' and col_type.ignored = false
+		join experiment_value col_name on col_name.experiment_state_id = es.id and col_name.ls_kind = 'column name' and col_name.ignored = false
+		join experiment_value col_units on col_units.experiment_state_id = es.id and col_units.ls_kind = 'column units' and col_units.ignored = false
+		left join experiment_value col_conc on col_conc.experiment_state_id = es.id and col_conc.ls_kind = 'column concentration' and col_conc.ignored = false
+		left join experiment_value col_conc_units on col_conc_units.experiment_state_id = es.id and col_conc_units.ls_kind = 'column conc units' and col_conc_units.ignored = false
+		left join experiment_value hide_col on hide_col.experiment_state_id = es.id and hide_col.ls_kind = 'hide column' and hide_col.ignored = false
+		left join experiment_value cond_col on cond_col.experiment_state_id = es.id and cond_col.ls_kind = 'condition column' and cond_col.ignored = false
+		left join experiment_value col_order on col_order.experiment_state_id = es.id and col_order.ls_kind = 'column order' and col_order.ignored = false
+		group by 
+			e.protocol_id,
+			coalesce(col_type.code_value, col_type.string_value),
+			coalesce(col_name.code_value, col_name.string_value),
+			coalesce(col_name.code_value, col_name.string_value),
+			col_conc.numeric_value,
+		 	coalesce(col_conc_units.code_value, col_conc_units.string_value),
+			coalesce(hide_col.code_value, hide_col.string_value),
+			coalesce(cond_col.code_value, cond_col.string_value)
+		) a;
+
+-- Define a function to create the protocol column order states and values
+-- this is nearly identical to the function above, just using protocol instead of experiment
+create or replace function create_protocol_data_column_order_state(protocol_id bigint, transaction_id bigint, column_order int, column_type varchar, column_name varchar, 
```

Comment:
Wondering if before this, we should fill in a `"strict endpoint matching"` value into the protocol metadata state and default it to `false`

Reply:
Yeah, I think you're right. I'll add that.

Reply:
Actually, I'm not going to. I think it's more accurate to be missing a value, and program the R & frontend code to use the default from the configs in cases where protocols are missing that value. If I bulk populate everything with `false` then it'll make setting the default strictMatch config to "true" pretty useless.